docs/source/programming.md

```diff
@@ -0,0 +1,123 @@
+- If the information is stored in the $\left|0\right>$ and $\left|1\right>$ states of the `XY` basis, $\hat{U}_{ij}(R_{ij}) =\frac{C_3}{R_{ij}^3} (|1\rangle\langle 0|_i |0\rangle\langle 1|_j + |0\rangle\langle 1|_i |1\rangle\langle 0|_j)$, where the interaction strength is $\frac{C_3}{R_{ij}^3}$, with $C_3$ a coefficient that depends on the energy levels used to encode $\left|0\right>$ and $\left|1\right>$. The entangling operator between atom $i$ and $j$ is $\hat{\sigma}_i^{+}\hat{\sigma}_j^{-} + \hat{\sigma}_i^{-}\hat{\sigma}_j^{+} = |1\rangle\langle 0|_i |0\rangle\langle 1|_j + |0\rangle\langle 1|_i |1\rangle\langle 0|_j$. This interaction hamiltonian is associated with the _XY Hamiltonian_ and is a less common mode of operation, usually accessible only in select neutral-atom devices.
+
+:::{important}
+With Pulser, you program the interaction Hamiltonian by setting the distance between atoms, $R_{ij}$, and the choice of eigenstates used in the computation and the Rydberg level(s) targeted by the device. Most commonly, the ground and Rydberg states ($\left|g\right>$ and $\left|r\right>$) are used, such that
+$$
+H^\text{int}_{ij} = \frac{C_6}{R_{ij}^6} \hat{n}_i\hat{n}_j
+$$
+When providing the distance between atoms, Pulser ensures that you respect the constraints of your chosen device.
+:::
+
+#### 2.4. Evolution of the system's state
+
+For a system of atoms in the initial state $\left|\Psi_0\right>$, the final state of the system after a time $\Delta t$ is:
+$$ \left|\Psi_f\right> = \exp\left(-\frac{i}{\hbar}\int_0^{\Delta t} H(t) dt\right)$$
+
+## Writing a Pulser program
+
+As outlined above, Pulser lets you program an Hamiltonian so that you can manipulate the quantum state of a system of atoms, that encode the states. The series of necessary instructions is encapsulated in the so-called Pulser `Sequence`. Here is a step-by-step guide to create your own Pulser `Sequence`.
+
+### 1. Pick a Device
+
+```{mermaid}
+
+flowchart TB
+  A[Picking a Device] --> B{Do I want to run on a QPU}
+  B -->|Yes| C[Pick the Device from the QPU backend]
+  B -->|No| D[Do I want to be constrained by all the QPU specs ?]
+  D -->|Yes| E[Use a `Device` (like `pulser.AnalogDevice`)]
+  D --> |No| F[Use a `VirtualDevice` (like `pulser.MockDevice`)]
+
+```
+
+The `Device` you select will dictate some parameters and constrain others. For instance, the value of the $C_6$ and $C_3$ coefficients of the interaction Hamiltonian are defined by the device. For a complete view of the constraints introduced by the device, [check its description](./apidoc/core.rst).
```

Comment:
Before the "For a complete view..." I would add a short but compelling sentence saying
"We strongly advice the user to have a look at the different device specifications to have a complete view of what can be achieved with Pulser. "

Reply:
What about adding this in the conclusion ? I don't want to scare anyone/make them visit the whole documentation at this stage of the recipe. Moreover, I currently don't have the page about the device done and I won't advice them to have a look at the API documentation at the first stage 😅

Reply:
good idea to have at the end.

imho you should write in terms of what we plan to have and not what we only currently have. Otherwise you'll have always to go back to what you have written and chances are we forget.
It is a suggestion, feel free to adopt the strategy you prefer of course. It is somehow related to the discussion we were having yesterday about having a clean state new doc or to implement changes on the fly in the currently existing version. I would opt for the first for the main reasons of making more easy to read and possibly share for intermediate feedbacks, but it's your call.